packages/compiler/src/server/type-signature.ts

```diff
@@ -149,6 +156,15 @@ function getModelSignature(type: Model, includeBody: boolean) {
     }
     return `${type.kind.toLowerCase()} ${getPrintableTypeName(type)}{\n${propDesc.map((d) => `${d};`).join("\n")}\n}`;
   } else {
+    if (useFormattedTemplateParameters && type.node) {
+      const formatted = await printTypeSpecNode(type.node);
+      // Extract the head of the model declaration (excluding body)
+      const match = formatted.match(/^(model\s+[^{]+)/);
```

Comment:
It would be better here that it only calls the `printTypeSpecNode` on the template parameter default node not the whole model.

Model is also not the only type that can be a template, this should be done generally for all template parmaeters

Reply:
also a bit unfortunate printTypeSpecNode is async and makes everything here async. Maybe as this is a workaround for a lack of exposition in the type graph anyway it would be better to just inject the node rawText (check how it is done in the printer.ts with `getRawText`)